internal/dataplane/kongstate/service_backend.go

```diff
@@ -0,0 +1,108 @@
+package kongstate
+
+import (
+	"errors"
+
+	"github.com/samber/lo"
+	"github.com/samber/mo"
+)
+
+// ServiceBackendType is the type of the backend.
+type ServiceBackendType string
+
+const (
+	// ServiceBackendTypeKongServiceFacade means that the backend is an incubatorv1alpha1.KongServiceFacade.
+	ServiceBackendTypeKongServiceFacade ServiceBackendType = "KongServiceFacade"
+
+	// ServiceBackendTypeKubernetesService means that the backend is a Kubernetes Service.
+	ServiceBackendTypeKubernetesService ServiceBackendType = "KubernetesService"
+)
+
+type ServiceBackends []ServiceBackend
+
+// ServiceBackend represents a backend for a Kong Service. It can be a Kubernetes Service or a KongServiceFacade.
+type ServiceBackend struct {
+	backendType ServiceBackendType
+	name        string
+	namespace   string
+	portDef     PortDef
+	weight      *int
+}
+
+// NewServiceBackend creates a new ServiceBackend with an arbitrary backend type.
+func NewServiceBackend(
+	t ServiceBackendType,
+	namespace string,
+	name string,
```

Comment:
nit - for your judge ;) order of `namespace` and `name` can be messed up because both have the same time. Using `NamespacedName` protects from such mistakes, but it makes it more verbose. I don't have strong preference. Just food for thought :)

Reply:
I agree - better safe than sorry. Changed functions to accept `NamespacedName` 👍 2288cdd